ui/app/src/components/RenameDashboardDialog/RenameDashboardDialog.tsx

```diff
@@ -0,0 +1,111 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { ChangeEvent, Dispatch, DispatchWithoutAction, useState } from 'react';
+import { Dialog, DialogTitle, DialogContent, DialogActions, Button, TextField, IconButton } from '@mui/material';
+import { DashboardResource } from '@perses-dev/core';
+import CloseIcon from 'mdi-material-ui/Close';
+import { dashboardDisplayName } from '@perses-dev/core/dist/utils/text';
+
+export interface RenameDashboardDialogProps {
+  open: boolean;
+  onClose: DispatchWithoutAction;
+  onConfirm: Dispatch<string>;
+  dashboard: DashboardResource;
+}
+
+/**
+ * Dialog used to rename a dashboard.
+ * @param props.open Define if the dialog should be opened or not.
+ * @param props.closeDialog Provides the function to close itself.
+ * @param props.onConfirm Action to perform when user confirmed.
+ * @param props.dashboard The dashboard resource to rename.
+ * @constructor
+ */
+const RenameDashboardDialog = (props: RenameDashboardDialogProps) => {
+  const { open, onClose, onConfirm, dashboard } = props;
+  const [name, setName] = useState<string>();
+  const [error, setError] = useState<string>();
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setName(e.target.value);
+    if (!e.target.value) {
+      setError('Required');
+    } else if (!/^[a-zA-Z0-9_.:-]+$/.exec(e.target.value)) {
+      setError('Special characters allowed are _ . : -');
+    } else {
+      setError(undefined);
+    }
+  };
+
+  // Reinitialize form for next time the dialog is opened
+  const resetForm = () => {
+    setName(undefined);
+    setError(undefined);
+  };
+
+  const handleConfirm = () => {
+    if (name) {
+      onClose();
+      onConfirm(name);
+      resetForm();
+    }
+  };
+
+  const handleClose = () => {
+    onClose();
+    resetForm();
+  };
+
+  return (
+    <Dialog open={open} onClose={handleClose} aria-labelledby="confirm-dialog">
+      <DialogTitle id="confirm-dialog">Rename the dashboard {dashboardDisplayName(dashboard)}</DialogTitle>
+      <IconButton
+        aria-label="Close"
+        onClick={handleClose}
+        sx={(theme) => ({
+          position: 'absolute',
+          top: theme.spacing(0.5),
+          right: theme.spacing(0.5),
+        })}
+      >
+        <CloseIcon />
+      </IconButton>
+      <DialogContent dividers sx={{ width: '500px' }}>
+        <TextField
+          required
+          margin="dense"
+          id="name"
+          label="Dashboard name"
```

Comment:
```suggestion
          label="Name"
```

Reply:
For consistency with other parts of the app

Reply:
I will do the change